processing/src/main/java/io/druid/segment/filter/SelectorFilter.java

```diff
@@ -31,25 +40,68 @@
 {
   private final String dimension;
   private final String value;
+  private final ExtractionFn extractionFn;
 
   public SelectorFilter(
       String dimension,
-      String value
+      String value,
+      ExtractionFn extractionFn
   )
   {
     this.dimension = dimension;
     this.value = value;
+    this.extractionFn = extractionFn;
   }
 
   @Override
   public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   {
-    return selector.getBitmapIndex(dimension, value);
+    if (extractionFn == null) {
+      return selector.getBitmapIndex(dimension, value);
+    } else {
+      final List<Filter> filters = makeFilters(selector);
+      if (filters.isEmpty()) {
+        return selector.getBitmapFactory().makeEmptyImmutableBitmap();
+      }
+      return new OrFilter(makeFilters(selector)).getBitmapIndex(selector);
```

Comment:
does makeFilters need to be called again here? can you just use `filters` from previously?